src/time_integration/methods_SSP.jl

```diff
@@ -0,0 +1,218 @@
+  sol::Sol # faked
+  f::F
+  alg::Alg
+  opts::SimpleIntegratorSSPOptions
+  finalstep::Bool # added for convenience
+end
+
+# Forward integrator.stats.naccept to integrator.iter (see GitHub PR#771)
+function Base.getproperty(integrator::SimpleIntegratorSSP, field::Symbol)
+  if field === :stats
+    return (naccept = getfield(integrator, :iter),)
+  end
+  # general fallback
+  return getfield(integrator, field)
+end
+
+"""
+    solve(ode; dt, callbacks, kwargs...)
+
+The following structures and methods provide a implementation of the third-order SSP Runge-Kutta
+method [`SimpleSSPRK33`](@ref).
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+function solve(ode::ODEProblem; alg=SimpleSSPRK33()::SimpleAlgorithmSSP,
+               dt, callback=nothing, kwargs...)
+  u = copy(ode.u0)
+  du = similar(u)
+  r0 = similar(u)
+  t = first(ode.tspan)
+  iter = 0
+  integrator = SimpleIntegratorSSP(u, du, r0, t, dt, zero(dt), iter, ode.p,
+                  (prob=ode,), ode.f, alg,
+                  SimpleIntegratorSSPOptions(callback, ode.tspan; kwargs...), false)
+
+  # resize container
+  resize!(integrator.p, nelements(integrator.p.solver, integrator.p.cache))
+
+  # initialize callbacks
+  if callback isa CallbackSet
+    for cb in callback.continuous_callbacks
+      error("unsupported")
+    end
+    for cb in callback.discrete_callbacks
+      cb.initialize(cb, integrator.u, integrator.t, integrator)
+    end
+  elseif !isnothing(callback)
+    error("unsupported")
+  end
+
+  solve!(integrator)
+end
+
+function solve!(integrator::SimpleIntegratorSSP)
+  @unpack indicator = integrator.p.solver.volume_integral
+  @unpack prob = integrator.sol
+  @unpack alg = integrator
+  t_end = last(prob.tspan)
+  callbacks = integrator.opts.callback
+
+  integrator.finalstep = false
+  @trixi_timeit timer() "main loop" while !integrator.finalstep
+    if isnan(integrator.dt)
+      error("time step size `dt` is NaN")
+    end
+
+    # if the next iteration would push the simulation beyond the end time, set dt accordingly
+    if integrator.t + integrator.dt > t_end || isapprox(integrator.t + integrator.dt, t_end)
+      integrator.dt = t_end - integrator.t
+      terminate!(integrator)
+    end
+
+    @. integrator.r0 = integrator.u
+    for stage in eachindex(alg.c)
+      @trixi_timeit timer() "Runge-Kutta stage" begin
+        t_stage = integrator.t + integrator.dt * alg.c[stage]
+        # compute du
+        integrator.f(integrator.du, integrator.u, integrator.p, t_stage)
+
+        # perform forward Euler step
+        @. integrator.u = integrator.u + integrator.dt * integrator.du
+      end
+      @trixi_timeit timer() "Antidiffusive stage" antidiffusive_stage!(integrator.u, t_stage, integrator.dt, integrator.p, indicator)
+
+      # perform convex combination
+      @. integrator.u = alg.a[stage] * integrator.r0 + alg.b[stage] * integrator.u
+    end
```

Comment:
Besides minor changes in the code, there is one major potential change. Currently, I always use the self-implemented SSPRK33 method for flexibility. Of course, it would be optimal if we could use the time integration algorithms of OrdinaryDiffEq.jl.
Therefore the question: Is it possible with OrdinaryDiffEq and possibly stage callbacks to perform an antidiffusive/limiting stage after the forward Euler step but before the convex combination? @ranocha @sloede 

So far, I have only found that the stage callback is performed after the entire stage.

Reply:
That's correct. With the current setup, what you would like to do is not possible.

Reply:
Okay too bad. Unfortunately, I don't see any other way with IDP limiting. Thank you anyway.